addons/mail/static/src/owl/store/messaging_actions.js

```diff
@@ -1877,6 +2083,26 @@ const actions = {
         });
         return threadCacheLocalId;
     },
+    /**
+     * @private
+     * Delete a thread from the store
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param threadLocalId the local id of the thread to delete
+     */
+    _deleteThread({ dispatch, state }, threadLocalId) {
+        const thread = state.threads[threadLocalId];
+        // Delete attachments
+        for (const attachmentLocalId of thread.attachmentLocalIds) {
+            dispatch('deleteAttachment', attachmentLocalId);
+        }
+        // Delete messages
+        for (const messageLocalId of thread.messageLocalIds) {
+            dispatch('deleteMessage', messageLocalId);
+        }
+        delete state.threads[threadLocalId];
```

Comment:
This doesn't look correct, but at least it only affects temporary threads at the moment, so some damage control!

An attachment can be linked to many threads, same for messages. That's why it doesn't make sense for deleting a thread to also delete related messages and attachments.
Also, deleting a thread should also delete causally related entities, like the composer of a thread and its caches.

Since this is hard to implement such an action for all kind of threads, I suggest to limit it for temporary threads.
(by the way, better modelling of entities and auto-compute of their links should handle that case, still work-in-progress, be patient ;) )

Reply:
Yes it's only for temporary threads for now and the idea here is to actually delete the thread (like the local pendant of removing from database) not "remove from store" so for me it makes sense that in that case all elements linked to it could/should be deleted :)

I can't wait for the "Kuhn-micro-ORM" :p !

Reply:
Shh... this acronym is a dirty word, remember? ;p

Reply:
Would still rename action to clarify this is for temporary threads.